### DISCUSSION TITLE
FrozenSnowFox

### DISCUSSION THREAD

**Discussion by RimWorldKorea** [RimWorldKorea/RMK#229] [일반]
Document: [XML Extensions](https://github.com/15adhami/XmlExtensions/wiki/Adding-Translations)

In the current state, translators have translated your mod using list nodes such as:
 * <FrozenSnowFoxTweaksSettings.settings.158.settings.0.text> or
 * <FrozenSnowFoxTweaksSettings.settings.172.settings.8.caseFalse.1.text> or ...

However, with this method, If any change are applied on mod, the entire addresses break, making translators to relocate all the affected nodes.

<hr>

<img width="278" alt="화면 캡처 2023-12-01 161500" src="https://github.com/RimWorldKorea/RMK/assets/150018627/57ef9120-79e0-447f-8732-b20d6f1cc056">

<p><p>With tKey, they can easily reference the texts that need to be translated, as shown below

<p><p><img width="317" alt="화면 캡처 2023-12-01 164312" src="https://github.com/RimWorldKorea/RMK/assets/150018627/80e3d0d4-7cdc-4c9b-99e5-8d663305bec8">

Therefore, if a part of the mod is modified, there will be no need to review the entire translation.

**Comment by RimWorldKorea** [comment]
@zilrty
>Ah, I see now. Basically a unique id you can reference instead of requiring things to be in a specific order. It looks easy enough, although perhaps a tad tedious since I've got so many tweaks now. I'll set up a couple later on and have you confirm its fine before doing more.

기다리던지 아님 여유있으면 도와주던지 하면 될 듯

**Reply by zilrty** [reply]
오우 모더한테 직접 물어봤구나

번역 번거로운건 자기도 알고 있네 ㅋㅋ

기다리면 잘해주것지

**Comment by RimWorldKorea** [comment]
![제목 없음](https://github.com/RimWorldKorea/RMK/assets/150018627/f04c82dc-7d8d-4eed-9ea9-dcac5cebcdd5)

**Comment by FrozenSnowFox** [comment]
I've gone through my mods and added tkey's for [FSF] FrozenSnowFox Tweaks, [FSF] Complex Jobs Remaster and [FSF] More Quests. Hopefully this makes things easier for you in the future. Thank you for taking the time to translate my mod.

Let me know if you spot any tkey's that don't follow the syntax I set or have typos. There were so many of them I wouldn't be surprised if I made a mistake somewhere.

**Reply by RimWorldKorea** [reply]
[Open with latest MS Excel.xlsx](https://github.com/RimWorldKorea/RMK/files/13536155/Open.with.latest.MS.Excel.xlsx)
Need to be updated together if there is change including tKey.
You can place this file somewhere within the mod.

**Reply by RimWorldKorea** [reply]
![제목 없음](https://github.com/RimWorldKorea/RMK/assets/150018627/ed5e0a30-c240-4391-abb3-446e017152e8)

**Comment by RimWorldKorea** [comment]
@zilrty 작업용 파일 만들어줄테니까 ㄱㄷ

**Reply by RimWorldKorea** [reply]
@zilrty [Translator Sheet.xlsx](https://github.com/RimWorldKorea/RMK/files/13533741/Translator.Sheet.xlsx)

일하시오

**Reply by zilrty** [reply]
핫산 핫산!

**Comment by zilrty** [comment]
@RimWorldKorea 

모더가 tkey 지정 안한곳이 한군데 있음

1.4\Royalty\Patches\aaaFSF_Royalty_Settings.xml 에 139번째 줄에 대한 tkey가 없음

일단 저거 뺀거 라도 올릴까?

**Reply by RimWorldKorea** [reply]
ㅇㅇ xlsx파일 포함해서

**Reply by zilrty** [reply]
[[FSF] FrozenSnowFox Tweaks - 2893432492.zip](https://github.com/RimWorldKorea/RMK/files/13536091/FSF.FrozenSnowFox.Tweaks.-.2893432492.zip)

**Reply by RimWorldKorea** [reply]
작업할때 뭐 썼음? 수식이 다 깨져있네
상관은 없긴 한데

**Reply by zilrty** [reply]
한셀로 한번 열어서 그런가
